tests/integration/error_warning_file_tests.py

```diff
@@ -57,6 +60,9 @@ class ErrorWarningTests(BaseTestValidator):
             submission: the submission foundation to be used for all the tests
             val_job: the validation job to be used for all the tests
     """
+    CHUNK_SIZES = [5]
+    PARALLEL_OPTIONS = [True, False]
+    CONFIGS = list(itertools.product(CHUNK_SIZES, PARALLEL_OPTIONS))
 
     @classmethod
```

Comment:
Create additional integration tests that test out what you expect to happen when one of many parallel processes has an exception
1. Are all other sibling processes in the pool that are working on chunks in parallel _also_ stopped when any one causes an error
1. Does the proper exception information bubble up to the sqs worker process, and get logged?
1. Do transactions all roll back and no data saved to the db (may need to run outside of the transactional-rollback test cases, and do your own setup teardown/cleanup for a test, with actual commits)
1. Do all processes get cleaned up: sibling pool processes, pool's parent worker process, worker's parent dispatcher process?
    - You can see if there are tests in the "reaping" code from USAspending. If not, may need to test this manually - kill things, then inspect running processes on the box with `ps -faux`

Reply:
Added a test which should cover these points:
1. To my knowledge, they are not stopped immediately (only solution to that would be either joining the processes as they come in which would be similar to the iterative load or using the `errored` flag I added to the shared_dict on every line of the async method). That being said, the flag I added prevents other loads from starting if it's already errored.
2. Code's been updated such that the proper exception information bubbles up to the parent process which should be carried by the SQS worker process.
3. With the redesign of the database work in the function, the proper table only gets updated on a successful load. The temp tables also get dropped in the `finally` of the loader.
4. With the test I've added, I've checked to make sure the child processes are cleaned up by the end of the function. The cleanup outside of the function has remained the same.